File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/ServiceContracts/Northwind.SupplierRow.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class SupplierRow
    {
        public Int32? SupplierID { get; set; }
        public String CompanyName { get; set; }
        public String ContactName { get; set; }
        public String ContactTitle { get; set; }
        public String Address { get; set; }
        public String City { get; set; }
        public String Region { get; set; }
        public String PostalCode { get; set; }
        public String Country { get; set; }
        public String Phone { get; set; }
        public String Fax { get; set; }
        public String HomePage { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string SupplierID = "SupplierID";
            [InlineConstant] public const string CompanyName = "CompanyName";
            [InlineConstant] public const string ContactName = "ContactName";
            [InlineConstant] public const string ContactTitle = "ContactTitle";
            [InlineConstant] public const string Address = "Address";
            [InlineConstant] public const string City = "City";
            [InlineConstant] public const string Region = "Region";
            [InlineConstant] public const string PostalCode = "PostalCode";
            [InlineConstant] public const string Country = "Country";
            [InlineConstant] public const string Phone = "Phone";
            [InlineConstant] public const string Fax = "Fax";
            [InlineConstant] public const string HomePage = "HomePage";
        }
    }
    
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.SupplierRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface SupplierRow {
    SupplierID?: number;
    CompanyName?: string;
    ContactName?: string;
    ContactTitle?: string;
    Address?: string;
    City?: string;
    Region?: string;
    PostalCode?: string;
    Country?: string;
    Phone?: string;
    Fax?: string;
    HomePage?: string;
}

export abstract class SupplierRow {
    static readonly idProperty = 'SupplierID';
    static readonly nameProperty = 'CompanyName';
    static readonly localTextPrefix = 'Northwind.Supplier';
    static readonly lookupKey = 'Northwind.Supplier';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<SupplierRow>('Northwind.Supplier') }
    static async getLookupAsync() { return getLookupAsync<SupplierRow>('Northwind.Supplier') }

    static readonly deletePermission = 'Northwind:General';
    static readonly insertPermission = 'Northwind:General';
    static readonly readPermission = 'Northwind:General';
    static readonly updatePermission = 'Northwind:General';

    static readonly Fields = fieldsProxy<SupplierRow>();
}